benchmarks/dynamic_config_envoy_proxy.py

```diff
@@ -68,17 +64,45 @@ def setUp(self):
     self._dynamic_config_settings.output_file = output_file
     logging.info(f"Injecting dynamic configuration. Output file: {output_file}")
 
-    # TODO(kbaichoo): we only hardcode a single endpoint, but will expand on this.
-    endpoints = self._dynamic_config_settings.clusters[0].endpoints.add()
-    endpoints.ip = self.test_server.server_ip
-    endpoints.port = self.test_server.server_port
+    # TODO(kbaichoo): Don't hardcode the clusters based on path configuration.
+    if '15_listeners' in self._nighthawk_test_config_path:
+      clusters = [
+          'cluster_one',
+          'cluster_two',
+          'cluster_three',
+          'cluster_four',
+          'cluster_five',
+      ]
+    else:
+      clusters = ['service_envoyproxy_io']
+
+    self._assignEndpointsToClusters(clusters)
 
     self._dynamic_config_controller = dynamic_config_server.DynamicConfigController(
         self._dynamic_config_settings)
 
     assert (self._dynamic_config_controller.start())
     logging.info("dynamic configuration running")
 
+  def _assignEndpointsToClusters(self, clusters):
+    """Process all backend endpoints, round robin assigning to the given clusters."""
+    available_endpoints = []
+
+    for endpoint in self.getAllTestServerRootUris():
+      ip_and_port = endpoint.split('/')[2]
+      available_endpoints.append(ip_and_port)
+
+    for cluster in clusters:
+      new_cluster = self._dynamic_config_settings.clusters.add()
+      new_cluster.name = cluster
+    cluster_idx = 0
+    for endpoint in available_endpoints:
+      new_endpoint = self._dynamic_config_settings.clusters[cluster_idx].endpoints.add()
+      ip, port = endpoint.rsplit(':', maxsplit=1)
```

Comment:
Can we extract the logic that parses the ip / port and the one that calculates the cluster index and unit test it to improve test coverage?

Reply:
extracted the utility for ip/port and added unit test for it